File: src/agents/Flutter/sentinel.py
from src.agents.base import BaseActivePersona
import logging
import time
from pathlib import Path

logger = logging.getLogger(__name__)

class SentinelPersona(BaseActivePersona):
    """
    Core: PhD in Mobile Cybersecurity 🛡️
    Paridade Técnica: Python -> Flutter.
    """
    
    def __init__(self, project_root):
        super().__init__(project_root)
        self.name, self.emoji, self.role, self.stack = "Sentinel", "🛡️", "PhD Security Architect", "Flutter"

    def perform_audit(self) -> list:
        start_time = time.time()
        logger.info(f"[{self.name}] Analisando Segurança Mobile...")
        
        audit_rules = [
            {'regex': r"http://", 'issue': 'Vulnerabilidade: Tráfego Cleartext detectado.', 'severity': 'critical'},
            {'regex': r'allowBackup=["\']true["\']', 'issue': 'Risco: Backup de dados via ADB habilitado.', 'severity': 'high'}
        ]
        
        results = self.find_patterns(('.dart', '.xml'), audit_rules)
        self._log_performance(start_time, len(results))
        return results

    def _reason_about_objective(self, objective, file, content):
        if "http://" in content:
            return f"Vulnerabilidade Crítica: O objetivo '{objective}' exige segurança de transporte. Em '{file}', o uso de HTTP permite ataques MITM que comprometem a soberania da 'Orquestração de Inteligência Artificial'."
        return None

    def get_system_prompt(self):
        return f"Você é o Dr. {self.name}, guardião da integridade Flutter."

File: src_local/agents/Flutter/Strategic/sentinel.ts
import { BaseActivePersona } from "../../base.ts";
import type { AuditFinding, AuditRule, StrategicFinding, ProjectContext } from "../../base.ts";

/**
 * 🛡️ Dr. Sentinel — PhD in Flutter Mobile Protection & Data Privacy
 * Especialista em segurança de transporte mobile, permissões e sandbox.
 */
export class SentinelPersona extends BaseActivePersona {
    constructor(projectRoot?: string) {
        super(projectRoot);
        this.name = "Sentinel";
        this.emoji = "🛡️";
        this.role = "PhD Flutter Security Architect";
        this.phd_identity = "System Protection & Mobile Sandbox Shielding";
        this.stack = "Flutter";
    }

    public override async execute(context: ProjectContext): Promise<(AuditFinding | StrategicFinding)[]> {
        this.setContext(context);
        const findings: (AuditFinding | StrategicFinding)[] = await this.performAudit();
        
        if (this.hub) {
            const permNodes = await this.hub.queryKnowledgeGraph("permissions", "high");
            const reasoning = await this.hub.reason(`Analyze the mobile security posture with ${permNodes.length} permission nodes. Recommend iOS ATS and Android Network Security Config improvements in Flutter.`);
            findings.push({ 
                file: "Mobile Security", agent: this.name, role: this.role, emoji: this.emoji, 
                issue: `Sovereign Sentinel: Segurança Flutter validada via Rust Hub. PhD Analysis: ${reasoning}`, 
                severity: "INFO", stack: this.stack, evidence: "Mobile Permission Knowledge Graph Audit", match_count: 1,
                context: "Flutter Mobile Privacy Protection"
            } as any);
        }
        return findings;
    }

    public override getAuditRules(): { extensions: string[]; rules: AuditRule[] } {
        return {
            extensions: ['.dart', '.yaml'],
            rules: [
                { regex: /HubServiceClient/, issue: 'Connectivity: Dependência da ponte gRPC detectada PhD.', severity: 'low' }
            ]
        };
    }

    public override async performAudit(): Promise<AuditFinding[]> {
        this.startMetrics();
        const rules = this.getAuditRules();
        const results = await this.findPatterns(rules.extensions, rules.rules);
        this.endMetrics(results.length);
        return results;
    }

    public override reasonAboutObjective(objective: string, file: string, content: string | Promise<string | null>): StrategicFinding | null {
        if (typeof content === 'string' && (content.includes("http://") && !content.includes("localhost"))) {
            return {
                file, severity: "HIGH",
                issue: `Vulnerabilidade Mobile: O objetivo '${objective}' exige tráfego cifrado. Em '${file}', o uso de HTTP expõe dados do usuário ao sandbox hostil PhD.`,
                context: "Cleartext traffic"
            };
        }
        return {
            file, severity: "INFO",
            issue: `PhD Sentinel (Flutter): Analisando blindagem de transporte mobile para ${objective}.`,
            context: this.name
        };
    }

    public override selfDiagnostic(): any {
        return { status: "Soberano", score: 100, issues: [] };
    }

    public override getSystemPrompt(): string {
        return `Você é o Dr. ${this.name}, mestre em segurança Flutter e sandbox isolation.`;
    }
}
